apis/smartweather.js

```diff
@@ -0,0 +1,471 @@
+			that.currentReport.AirFirmware = message.firmware_revision;
+			that.currentReport.ObservationTime = moment.unix(message.obs[0][0]).format('HH:mm:ss');
+			that.currentReport.AirPressure = message.obs[0][1];
+			if (that.currentReport.TemperatureSensorFail == 1)
+				that.currentReport.Temperature = 0;
+			else
+				that.currentReport.Temperature = message.obs[0][2];
+			if (that.currentReport.HumiditySensorFail == 1) {
+				that.currentReport.Humidity = 0;
+				that.currentReport.DewPoint = 0;
+			}
+			else {
+				that.currentReport.Humidity = message.obs[0][3];
+				that.currentReport.DewPoint = wformula.kelvinToCelcius(wformula.dewPointMagnusFormula(
+					wformula.celciusToKelvin(that.currentReport.Temperature), 
+					that.currentReport.Humidity));
+			}
+			that.currentReport.TemperatureApparent = wformula.kelvinToCelcius(wformula.australianAapparentTemperature(
+					wformula.celciusToKelvin(that.currentReport.Temperature),
+					that.currentReport.Humidity,
+					that.currentReport.WindSpeed));
+			that.currentReport.TemperatureMin = (that.currentReport.Temperature < that.currentReport.TemperatureMin) ?
+					that.currentReport.Temperature : that.currentReport.TemperatureMin;
+			that.currentReport.TemperatureWetBulb = 
+					this.getWetBulbTemperature(that.currentReport.Temperature, that.currentReport.Humidity);
+			that.currentReport.LightningStrikes = message.obs[0][4];
+			that.currentReport.LightningAvgDistance = message.obs[0][5];
+			that.currentReport.AirSensorBatteryLevel = this.getBatteryPercent(message.obs[0][6]);
+		}
+
+		if (message.type == 'obs_sky') {
+			that.currentReport.SkySerialNumber = message.serial_number;
+			that.currentReport.ObservationStation = that.currentReport.SkySerialNumber;
+			that.currentReport.SkyFirmware = message.firmware_revision;
+			that.currentReport.ObservationTime = moment.unix(message.obs[0][0]).format('HH:mm:ss');
+			if (that.currentReport.LightLevelSensorFail == 1)
+				that.currentReportLightLevel = 0;
+			else
+				that.currentReport.LightLevel = message.obs[0][1];
+			that.currentReport.UVIndex = message.obs[0][2];
+			that.currentReport.Rain1h = this.getHourlyAccumulatedRain(message.obs[0][0], message.obs[0][3]);
+		
+			// Use wind values reported by rapid_wind as they are instantanous,
+			// whereas obs_sky are averaged values over last minute
+			//that.currentReport.WindSpeed = message.obs[0][5];
+			//that.currentReport.WindDirection = converter.getWindDirection(message.obs[0][7]);
+			
+			// Wind values are min/max over last minute
+			that.currentReport.WindSpeedLull = message.obs[0][4];
+			that.currentReport.WindSpeedMax = message.obs[0][6];
+			
+			that.currentReport.SkySensorBatteryLevel = this.getBatteryPercent(message.obs[0][8]);
+			that.currentReport.SolarRadiation = message.obs[0][10];
+			// currentReport.RainDay = message.obs[0][11];
+			// Note that Local Day Rain Accumulation (Field 11) is always null as it is calculate at WeatherFlow
```

Comment:
Interesting … so this is kind of a lie in the current docs at https://weatherflow.github.io/Tempest/api/udp/v171/ then, because that indeed lists this field, but simultaneously shows that this is `null` in the example 🤷 Very odd.

I think we can clarify this:
```suggestion
			// Note that Local Day Rain Accumulation (Field 11) is currently always null. Hence we have to approximate it with data available to us.
```